ui/src/lib/ButtonDelete.svelte

```diff
@@ -0,0 +1,22 @@
+<script lang="ts">
+  import SvgIcon from "./SvgIcon.svelte";
+
+  export let onDelete: (event: MouseEvent) => void;
+  export let moreClasses: string;
+
+  function handleClick(event: MouseEvent) {
+    // It ensures that the click doesn't bubble up to parent elements
+    // It's important when the button is nested inside other clickable elements
+    event.stopPropagation();
+    onDelete(event);
+  }
+</script>
+
+<button
+  type="button"
+  on:click={handleClick}
```

Comment:
```suggestion
  on:click|stopPropagation
```

Then remove the 'onDelete' prop and handle it as an on:click event from the parent component